TombEngine/Shaders/Math.hlsli

```diff
@@ -121,6 +125,42 @@ float SimplexNoise(float3 v)
 									 dot(p2,x2), dot(p3,x3) ) );
 }
 
+float Noise2D (float2 input)
+{
+	float2 i = floor(input);
+	float2 f = frac(input);
+
+	// Four corners in 2D of a tile
+	float a = RandomValue(i);
+	float b = RandomValue(i + float2(1.0, 0.0));
+	float c = RandomValue(i + float2(0.0, 1.0));
+	float d = RandomValue(i + float2(1.0, 1.0));
+
+	float2 u = f * f * (3.0 - 2.0 * f);
+
+	return lerp(a, b, u.x) +
+							(c - a) * u.y * (1.0 - u.x) +
+							(d - b) * u.x * u.y;
+}
+
+#define OCTAVES 6
+float fractalNoise (float2 input)
+{
+	// Initial values
+	float value = 0.0;
+	float amplitude = .5;
+	float frequency = 0.;
+	//
+	// Loop of octaves
+	for (int i = 0; i < OCTAVES; i++)
+		{
```

Comment:
Please format properly, according to coding conventions.

Reply:
Solrry